smarts/core/chassis.py

```diff
@@ -176,12 +188,23 @@ def bullet_id(self) -> str:
         return self._bullet_body._bullet_id
 
     @property
-    def speed(self):
+    def speed(self) -> float:
+        # in m/s
         return self._speed
 
     @property
     def velocity_vectors(self):
-        return (None, None)
+        # linear_velocity in m/s, angular_velocity in rad/s
+        vh = radians_to_vec(self.pose.heading)
+        if self._speed is not None:
+            linear_velocity = np.array((vh[0] * self._speed, vh[1] * self._speed, 0.0))
```

Comment:
Likely less optimised but a bit easier to look at:
```suggestion
            linear_velocity = np.array((vh[0], vh[1], 0.0)) * self._speed
```